firmware/common/operacake.c

```diff
@@ -166,10 +229,14 @@ uint8_t operacake_set_ports(uint8_t address, uint8_t PA, uint8_t PB) {
 
 	pa = port_to_pins(PA);
 	pb = port_to_pins(PB);
-		
+
 	reg = (OPERACAKE_GPIO_DISABLE | side
 					| pa | pb | OPERACAKE_EN_LEDS);
 	operacake_write_reg(oc_bus, address, OPERACAKE_REG_OUTPUT, reg);
+
+	// Keep track of manual settings for when we switch in/out of time/frequency modes.
+	operacake_boards[address].PA = PA;
+	operacake_boards[address].PB = PB;
```

Comment:
Do we want operacake_set_ports() to behave differently if it is called while in time or frequency mode? It gets called by usb_vendor_request_operacake_set_ports() which maybe should stall if we're in one of those modes? Or maybe it should force the mode to manual?

Reply:
I'm not sure about making a special case for set_ports / manual mode, I think it would be best to keep it consistent with the other modes, and I don't think we should automatically switch mode when setting dwell times or frequency ranges. I think it's reasonable to treat set_ports as a setting that only applies in manual mode, as long as we document that well.